query-engine/js-connectors/js/neon-js-connector/src/neon.ts

```diff
@@ -127,7 +140,11 @@ class PrismaNeon implements Connector, Closeable {
     const { sql, args: values } = query
 
     try {
-      return await this.client.query(sql, values)
+      if (this._httpMode) {
+        return await this.client!.query(sql, values)
+      } else {
+        return await this.httpClient!(sql, values)
+      }
```

Comment:
We can remove the need for these `!` TypeScript assertions by merging the "mode" into the same new object containing the actual `client`.

We could have something like:

```typescript
if (this.driver.mode === 'http') {
  // this.driver.client is inferred as NeonQueryFunction<false, true>
  return await this.driver.client(sql, values)
} else {
  // this.driver.client is inferred as Client
  return await this.driver.client.query(sql, values)
}
```